src/dmd/expressionsem.d

```diff
@@ -5485,12 +5488,9 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                      */
                     assert(tded.ty == Tfunction);
                     auto tdedf = tded.isTypeFunction();
-                    size_t dim = tdedf.parameterList.length;
                     auto args = new Parameters();
-                    args.reserve(dim);
```

Comment:
This removal of `reserve` is another little functional change. Not sure if it's worth to pre-iterate once over the list.